File: frontend/src/app/core/api/models/ctf.ts
/* tslint:disable */
/* eslint-disable */
import { Links } from '../models/links';
import { UserInfo } from '../models/user-info';
export interface Ctf {
  '_links'?: Links;
  author?: UserInfo;
  description?: string;
  filePath?: string;
  name?: string;
}

File: backend/src/main/java/dev/coll/ctf/domain/ctf/model/Ctf.java
package dev.coll.ctf.domain.ctf.model;

import java.util.HashSet;
import java.util.Set;

import dev.coll.ctf.domain.user.model.User;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

@Data
@Builder
@AllArgsConstructor
public class Ctf {
  private String name;
  private String description;
  private User author;
  private String flag;
  private String filePath;
  @ToString.Exclude
  @EqualsAndHashCode.Exclude
  @Builder.Default
  private Set<Solve> solves = new HashSet<>();
}

